src/main/java/net/kyori/adventure/text/minimessage/MiniMessageParser.java

```diff
@@ -239,8 +239,11 @@
 
     context.root(root);
     final Component comp = this.parse(root);
-    // at the end, take a look if we can flatten the tree a bit
-    return this.flatten(comp);
+    if (context.compactingMethod() != null) {
```

Comment:
I think it would be better if this was non-null, can always use an identity function as a default.

Reply:
I think i made an improvement????